controller/cache/cache.go

```diff
@@ -400,9 +400,10 @@ func (c *liveStateCache) getCluster(server string) (clustercache.ClusterCache, e
 		clustercache.SetPopulateResourceInfoHandler(func(un *unstructured.Unstructured, isRoot bool) (interface{}, bool) {
 			res := &ResourceInfo{}
 			populateNodeInfo(un, res)
+			cacheSettings := c.cacheSettings
```

Comment:
I was wondering, do I need to acquire a lock on the `c.lock` mutex here?

Reply:
Isn't it happening already on line 374?

Reply:
Yes, but this is a call-back function, which will be called outside the scope of the lock on line 374

Reply:
Also, looking at the code between lines 365 and 368, the assignment of `c.cacheSettings` is also done in a locked way. So probably makes sense to lock it here as well, just for the assignment to `cacheSettings`. 

However, since the `PopulateResourceInfoHandler` will be called on every resource on the cluster, I'm wondering about the performance impact of this lock.

Reply:
Mutexes break my brain. What problem would the mutex solve?

Reply:
@crenshaw-dev `c.cacheSettings` is a variable of composite type. The mutex would ensure exclusive operation on that variable for the current holder of the lock on the mutex, to ensure that no other goroutine would modify any of the fields in that struct at the same time we copy it (given that the other goroutine also tries to acquire a lock on the same mutex).

For example, `c.cacheSettings` is modified whenever there are changes to the Argo CD settings.

Reply:
That much I follow. But I'm trying to understand what the lock would span and what problem it would prevent. I'm not sure why we'd care if `c.cacheSettings` changed in the context of one of these callback executions.

Reply:
The lock would span the copy operation of assigning `c.cacheSettings` to the local variable `cacheSettings`, to perform the copy as one atomic operation.

Maybe in the current context, this doesn't seem to be important. However, when the cache settings get expanded at some point in time, and another process modifies the settings while we copy it, we may end up with non-working data.

Reply:
Makes sense, thanks!

Reply:
Thanks, I added the read locks.